src/plugins/discover/public/application/components/discover_grid/get_render_cell_value.tsx

```diff
@@ -51,20 +52,76 @@ export const getRenderCellValueFn = (
     return <span>-</span>;
   }
 
+  if (useNewFieldsApi && !field && row && row.fields) {
+    const innerColumns = Object.fromEntries(
+      Object.entries(row.fields as Record<string, unknown[]>).filter(([key]) => {
+        return key.indexOf(`${columnId}.`) === 0;
+      })
+    );
+    if (isDetails) {
+      // nicely formatted JSON for the expanded view
+      return <span>{JSON.stringify(innerColumns, null, 2)}</span>;
+    }
+
+    // Put the most important fields first
+    const highlights: Record<string, unknown> = (row.highlight as Record<string, unknown>) ?? {};
+    const highlightPairs: Array<[string, string]> = [];
+    const sourcePairs: Array<[string, string]> = [];
+    Object.entries(innerColumns).forEach(([key, values]) => {
+      const subField = indexPattern.getFieldByName(key)!;
+      const formatter = indexPattern.getFormatterForField(subField);
+      const formatted = (values as unknown[])
+        .map((val: unknown) =>
+          formatter.convert(val, 'html', {
+            field: subField,
+            hit: row,
+            indexPattern,
+          })
+        )
+        .join(', ');
+      const pairs = highlights[key] ? highlightPairs : sourcePairs;
+      pairs.push([key, formatted]);
+    });
+
+    return (
+      <EuiDescriptionList type="inline" compressed className="dscDiscoverGrid__descriptionList">
+        {[...highlightPairs, ...sourcePairs].map(([key, value]) => (
+          <Fragment key={key}>
+            <EuiDescriptionListTitle>{key}</EuiDescriptionListTitle>
+            <EuiDescriptionListDescription
+              dangerouslySetInnerHTML={{ __html: value }}
+              className="dscDiscoverGrid__descriptionListDescription"
+            />
+          </Fragment>
+        ))}
+      </EuiDescriptionList>
+    );
+  }
+
   if (field && field.type === '_source') {
     if (isDetails) {
       // nicely formatted JSON for the expanded view
       return <span>{JSON.stringify(row, null, 2)}</span>;
     }
     const formatted = indexPattern.formatHit(row);
 
+    // Put the most important fields first
```

Comment:
Sorting the field list is something that Discover has always done with highlighting, so this restores the behavior @kertal

Reply:
Wasn't aware of this details thanks a lot @wylieconlon!